### PR TITLE
MDTZ-1089: Refactor Figma webhook error handling (PR 2/3)

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
## Changes

- **refactor:** Replaces `FigmaWebhookService` with `figmaWebhookAuthMiddleware` for authenticating Figma webhook events.
- **refactor:** Deletes `ValidationError` and related error handling.
- **refactor:** Simplifies and improves schema definition for Figma events by defining separate schemas for each event type using [JSON Schema Discriminator](https://ajv.js.org/json-schema.html#discriminator).
- **test:** Resolves test flakiness for the `checkAuth` endpoints (caused by the usage of time-bound JWT tokens).

### CODE REVIEW

File: src/infrastructure/repositories/figma-team-repository.ts [hunk 1]
```diff
@@ -44,15 +44,17 @@ export class FigmaTeamRepository {
...
 
-	getByWebhookId = async (webhookId: string): Promise<FigmaTeam> => {
-		const dbModel = await prismaClient
-			.get()
-			.figmaTeam.findFirst({ where: { webhookId } });
+	/**
+	 * @internal
+	 * Required for tests only.
+	 */
+	get = async (id: string): Promise<FigmaTeam> => {
```
Comment (akostevich-atlassian) [inline src/infrastructure/repositories/figma-team-repository.ts:51]: **note:** Add `get` for simplifying integration tests.

File: src/web/routes/admin/auth/integration.test.ts [hunk 2]
```diff
@@ -75,44 +75,6 @@ describe('/admin/auth', () => {
 				.expect({ authorized: true });
 		});
```
Comment (akostevich-atlassian) [inline src/web/routes/admin/auth/integration.test.ts:77]: **note:** Add new integration tests, reorder existing ones and resolve test flakiness (caused by a time-bound JWT token).

File: src/web/routes/auth/integration.test.ts [hunk 3]
```diff
@@ -70,41 +70,6 @@ describe('/auth', () => {
 				.expect({ type: '3LO', authorized: true });
 		});
 
-		it('should return a response indicating that user is not authorized if no credentials stored', async () => {
```
Comment (akostevich-atlassian) [inline src/web/routes/auth/integration.test.ts:73]: **note:** Add new integration tests, reorder existing ones and resolve test flakiness (caused by a time-bound JWT token).

File: src/web/routes/figma/schemas.ts [hunk 4]
```diff
@@ -1,89 +1,50 @@
...
- * @see https://github.com/ajv-validator/ajv/issues/1664#issuecomment-873613644
- */
-const optionalNonNullable = <T>(schema: T): T & { nullable: true } =>
-	schema as T & { nullable: true };
+import type { JSONSchemaTypeWithId } from '../../../common/schema-validation';
 
-export const FIGMA_WEBHOOK_EVENT_REQUEST_SCHEMA: JSONSchemaTypeWithId<{
-	body: FigmaWebhookEventPayload;
-}> = {
+export const FIGMA_WEBHOOK_EVENT_REQUEST_SCHEMA = {
```
Comment (akostevich-atlassian) [inline src/web/routes/figma/schemas.ts:8]: **note:** Improve schema definition by using [JSON Schema Discriminator](https://ajv.js.org/json-schema.html#discriminator) (e.g., `file_key` gets required for the `FILE_UPDATE` event but is still optional for `PING`).

File: src/web/routes/figma/types.ts [hunk 5]
```diff
@@ -1,28 +1,42 @@
-import type { Request } from 'express';
+import type { Request, Response } from 'express';
 
-import type { FigmaWebhookEventType } from '../../../domain/entities';
+import type { FigmaTeam } from '../../../domain/entities';
 
-export type FigmaWebhookEventPayload = {
-	readonly event_type: FigmaWebhookEventType;
```
Comment (akostevich-atlassian) [inline src/web/routes/figma/types.ts:6]: **note:** Remove fields the app is not interested in.

**Review by stannnous — APPROVED** [review]

**Pull request merged**